server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/qualitymodel/DebtRatingGrid.java

```diff
@@ -42,20 +49,37 @@ public DebtRatingGrid(Configuration config) {
       for (int i = 0; i < 4; i++) {
         gridValues[i] = Double.parseDouble(grades[i]);
       }
+      this.gridFunctions = buildGridFunctions(gridValues);
     } catch (Exception e) {
       throw new IllegalArgumentException("The rating grid is incorrect. Expected something similar to '"
         + RATING_GRID_DEF_VALUES + "' and got '" + config.get(RATING_GRID).orElse("") + "'", e);
     }
   }
 
-  public Rating getRatingForDensity(double density) {
-    for (Rating rating : Rating.values()) {
-      double lowerBound = getGradeLowerBound(rating);
-      if (density >= lowerBound) {
-        return rating;
-      }
-    }
-    throw MessageException.of("The rating density value should be between 0 and " + Double.MAX_VALUE + " and got " + density);
+  public DebtRatingGrid(double[] gridValues) {
+    this.gridValues = Arrays.copyOf(gridValues, gridValues.length);
+    this.gridFunctions = buildGridFunctions(gridValues);
+    ;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this empty statement. ([squid:EmptyStatementUsageCheck](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fjl%2FSONAR-10193_fix_debt_computation&issues=AWC8QPVrqDrCo5ysgnIF&open=AWC8QPVrqDrCo5ysgnIF)

[ISSUE_KEY]: AWC8QPVrqDrCo5ysgnIF